portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,22 +38,29 @@
   private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
   private static final String TIME = "time";
   private static final String NEWEST_MESSAGE = "newest-message";
+  private static final String COMMENT_LIMIT = "comment-limit";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String limitString = request.getParameter("numComments");
+    int limit = convertToInt(limitString);
+
     Query query = new Query("Message").addSort("time", SortDirection.DESCENDING);
 
     PreparedQuery results = datastore.prepare(query);
 
     List<String> messages = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
-      String fullMessage = (String) entity.getProperty("newest-message");
-      long timestamp = (long) entity.getProperty("time");
+      String fullMessage = (String) entity.getProperty(NEWEST_MESSAGE);
+      long timestamp = (long) entity.getProperty(TIME);
 
       messages.add(fullMessage);
     }
 
-    String json = gson.toJson(messages);
+    List<String> limitedList =
```

Comment:
Instead of reversing the list here, would you want it the SortDirection to be ASCENDING in Loc 41?

Reply:
For clarification, you are saying change the sort direction for the sorting of the query?

Reply:
Yep! You can also add a TODO for this, to unblock the PR.